source/Octopus.Tentacle/Scripts/ScriptWorkspaceFactory.cs

```diff
@@ -143,8 +143,20 @@ public List<IScriptWorkspace> GetUncompletedWorkspaces()
 
         static bool IsUncompletedWorkspaceDirectory(string workspaceDirectory)
         {
+            //first we look for an output log. If there is one, the workspace is uncompleted 
             var outputLogFilePath = ScriptWorkspace.GetLogFilePath(workspaceDirectory);
-            return File.Exists(outputLogFilePath);
+            if (File.Exists(outputLogFilePath))
```

Comment:
Seems a bit unfortunate to have this check both here and in the workspace cleaner 🤔

Reply:
yeah, I think the check in the workspace cleaner is a bit of a fallback option for edge cases